cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -205,7 +206,8 @@ class S3TestMixin : public AwsTestMixin {
     ARROW_ASSIGN_OR_RAISE(minio_, GetMinioEnv()->GetOneServer());
     client_config_.reset(new Aws::Client::ClientConfiguration());
     client_config_->endpointOverride = ToAwsString(minio_->connect_string());
-    client_config_->scheme = Aws::Http::Scheme::HTTP;
+    client_config_->scheme = Aws::Http::Scheme::HTTPS;
+    client_config_->verifySSL = false;
```

Comment:
Could you set `caFile` instead?

Reply:
ok, i set the capath here.

Reply:
@kou  According to https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/client-config.html
the capath/cafile solution only work for linux, but not for Windows and MacOS, 
we may still need to set the verifySSL  to false for the test case.

Reply:
Oh...
OK. Could you use `#if defined(_WIN32) || defined(__APPLE__)` instead of `#ifdef _WIN32`?
This code may be used on macOS.